### PR TITLE
Reenable assert sanity testing with ADInplaceOrView reenable

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #88102

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: torch/_subclasses/fake_tensor.py [hunk 1]
```diff
@@ -720,7 +720,11 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
 
         # Some attribute queries that can be serviced directly
         # See Note [is_coalesced is dispatched]
-        if func in [torch.ops.aten.is_coalesced.default]:
+        if func in [
```
Comment (albanD) [inline torch/_subclasses/fake_tensor.py:723]: nit: should we get in the habit to use sets for these things? Even for the test skips, we're pretty inconsistent and use large-ish lists sometimes.

File: torch/_subclasses/meta_utils.py [hunk 2]
```diff
@@ -262,63 +264,75 @@ def is_c_of_r(complex_dtype, real_dtype):
...
-                                r = base.as_strided(
-                                    sizes, strides, sym(t.storage_offset())
-                                )
+                    # In some situations, MetaConverter may be called in a
+                    # context where autograd is disabled.  For the _is_view
+                    # assert to pass, we have to setup the autograd view
+                    # metadata anyway.  Do this by reenabling the
+                    # ADInplaceOrView key.  This is kind of a hack.
+                    old_exclude = torch._C._dispatch_tls_is_dispatch_key_excluded(torch._C.DispatchKey.ADInplaceOrView)
+                    #torch._C._dispatch_tls_set_dispatch_key_excluded(torch._C.DispatchKey.ADInplaceOrView, False)
```
Comment (albanD) [inline torch/_subclasses/meta_utils.py:273]: This shouldn't be commented?

**Review by albanD — APPROVED** [review]
SGTM

File: torch/_subclasses/fake_tensor.py (hunk 1 above)
Comment (ezyang) [reply torch/_subclasses/fake_tensor.py:723]: According to SO sets are always faster than lists, so yes, let's do it

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**